xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -243,6 +254,9 @@ func (b *clusterImplBalancer) updateClientConnState(s balancer.ClientConnState)
 
 	b.config = newConfig
 
+	b.inhibitPickerUpdates = true
+	defer func() { b.inhibitPickerUpdates = false }()
```

Comment:
Both update state and `UpdateClientConnState` are using the same serializer. So when the `updateClientConnState` callback is being executed in the serializer, not other callback (including the one scheduled by `UpdateState`) can be executed until `updateClientConnState` has returned. So inhibiting updates seems redundant.

Reply:
Arjan is right. Since the same serializer is used to handle updates from the parent and handle updates from the child, we will not be able to handle updates from the child until we return from `updateClientConnState`. This means that we need to refactor things out a little bit.

The `childState` field is the only shared field that is accessed from `UpdateState`. So, I think we might have to do the following:
- add a mutex that guards access to `childState` and `inhibitPickerUpdates`, and change existing methods to grab this mutex when they need to access these fields
- do not run `UpdateState` in the serializer, run it inline

Let me know what you think. Please don't take my word as is. Please ensure that the above approach would actually work.

Reply:
Thanks @easwars for your thoughts. I agree that currently since `UpdateState`, `UpdateClientConnState` and few other methods are using the same serializer, it will be executed one after the other. And hence child config can't be handled before `updateClientConnState`. 
I agree that we could make the `UpdateState` asynchronous by running it inline and not use serializer for it.

Reply:
Since you haven't assigned the PR back to me, I'm assuming that you are still working on it. I skimmed through it and it was still not going what I expect it to do.